api/v1/attributes.proto

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package istio.mixer.v1;
+
+option go_package="istio.io/mixer/api/v1;mixerpb";
+
+// Contains a set of attributes related to an event.
+message Attributes {
```

Comment:
I think the mixer code shouldn't be too ugly for this, it can even use the protobuf descriptor and do it fully generically, if the field names follow a standard pattern. Basically take field name, replace '_' with '.', and stick those all in a map, then all the rest of the code just uses that. No need for annotations, unless we need the description in code somehow.

This is meant to serve two purposes:
1. optimization
1. provide a set of well-known, well-typed attributes that proxies and mixers can agree on.

I don't think it premature for the second, but agree it may be for the first, which should make us careful to only add fields we think will live for 5+ years.